pyvex/block.py

```diff
@@ -69,14 +71,64 @@ def __init__(self, data, mem_addr, arch, max_inst=None, max_bytes=None, bytes_of
         self._size = None
 
         if data is not None:
-            lift(self, data, max_bytes, max_inst, bytes_offset, opt_level, traceflags)
+            lift(self, arch, mem_addr, data, max_bytes, max_inst, bytes_offset, opt_level, traceflags)
+
+    @staticmethod
+    def emptyBlock(arch, addr, statements=None, nxt=None, tyenv=None, jumpkind=None, direct_next=None, size=None):
+        block = IRSB(None, addr, arch)
+        block._set_attributes(statements, nxt, tyenv, jumpkind, direct_next)
+        return block
+
+    def copy(self):
+        return copy.deepcopy(self)
+
+    def extend(self, extendwith):
+        conversion_dict = { }
+        invalid_vals = (0xffffffff, -1)
+
+        def convert_tmp(tmp):
+            if tmp not in conversion_dict:
+                tmp_type = extendwith.tyenv.lookup(tmp)
+                conversion_dict[tmp] = self.tyenv.add(tmp_type)
+            return conversion_dict[tmp]
+
+        def convert_expr(expr):
+            if isinstance(expr, RdTmp):
+                expr.tmp = convert_tmp(expr.tmp)
+
+        for stmt in extendwith.statements:
+            if isinstance(stmt, WrTmp):
+                stmt.tmp = convert_tmp(stmt.tmp)
+            elif isinstance(stmt, LoadG):
+                stmt.dst = convert_tmp(stmt.dst)
+            elif isinstance(stmt, LLSC):
+                stmt.result = convert_tmp(stmt.result)
+            elif isinstance(stmt, Dirty):
+                if stmt.tmp not in invalid_vals:
+                    stmt.tmp = convert_tmp(stmt.tmp)
+                for e in stmt.args:
+                    convert_expr(e)
+            elif isinstance(stmt, CAS):
+                if stmt.oldLo not in invalid_vals: stmt.oldLo = convert_tmp(stmt.oldLo)
+                if stmt.oldHi not in invalid_vals: stmt.oldHi = convert_tmp(stmt.oldHi)
+            elif isinstance(stmt, LLSC):
+                stmt.result = convert_tmp(stmt.result)
+            for e in stmt.expressions:
+                convert_expr(e)
+            self.statements.append(stmt)
+        convert_expr(extendwith.next)
+        self.next = extendwith.next
+        self.jumpkind = extendwith.jumpkind
 
     def pp(self):
         """
         Pretty-print the IRSB to stdout.
         """
         print(self._pp_str())
 
+    def __str__(self):
```

Comment:
this'll be obnoxious unless you add a `__repr__` which prints a single-line description